internal/pkg/node/transformers.go

```diff
@@ -23,13 +24,96 @@ Populates a NodeConf struct from a NodeInfo, with the combined
 values from the underlying entries using Get.
 */
 func (nodeConf *NodeConf) GetFrom(nodeInfo NodeInfo) {
-	nodeConf.getterFrom(nodeInfo, (*Entry).Get, (*Entry).GetSlice)
+	recursiveGetter(&nodeInfo, nodeConf, (*Entry).Get, (*Entry).GetSlice)
+}
+
+/*
+Abstract function which populates a NodeConf from the given NodeInfo
+via getter functions. Calls recursive itself for nested structures.
+Panics if the NodeConf has fields which are not type of string,[]string,map[string]*ptr
+*/
+func recursiveGetter(
```

Comment:
Can you add some tests to demonstrate that recursiveGetter works correctly?

Reply:
Really, I think recursiveGetter and recursiveSetter (later) are symptoms of the whole Conf vs Entry dichotomy being much too complicated. I have some ideas for how to fix that; but, for now, these methods end up being huge and complex, and I think they need additional test functions in the suite to make sure they're behaving the way they're intended to.